groups/validate.go

```diff
@@ -17,29 +17,43 @@ limitations under the License.
 package main
 
 import (
-	"flag"
 	"fmt"
+	"log"
 	"os"
 	"path/filepath"
 	"reflect"
 	"sort"
 	"strings"
-	"testing"
 	"unicode/utf8"
 
 	"k8s.io/apimachinery/pkg/util/sets"
 )
 
-var cfg GroupsConfig
-var rConfig RestrictionsConfig
+func Validate(configPath, restrictionsPath, groupsPath *string) {
+	var err error
 
-var groupsPath = flag.String("groups-path", "", "Directory containing groups.yaml files")
-var restrictionsPath = flag.String("restrictions-path", "", "Path to the configuration file containing restrictions")
+	// Check if config.yaml is loadable
+	if *configPath != "" && !filepath.IsAbs(*configPath) {
+		fmt.Printf("config \"%s\" must be an absolute path\n", *configPath)
+		os.Exit(1)
+	}
 
-func TestMain(m *testing.M) {
-	flag.Parse()
-	var err error
+	if *configPath == "" {
+		baseDir, err := os.Getwd()
+		if err != nil {
+			fmt.Printf("Cannot get current working directory: %v\n", err)
+			os.Exit(1)
+		}
+		cPath := filepath.Join(baseDir, defaultConfigFile)
+		configPath = &cPath
+	}
 
+	if err := config.Load(*configPath, false); err != nil {
+		fmt.Printf("Could not load main config: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Check if restrictions.yaml is loadable
 	if *restrictionsPath != "" && !filepath.IsAbs(*restrictionsPath) {
 		fmt.Printf("restrictions-path \"%s\" must be an absolute path\n", *restrictionsPath)
 		os.Exit(1)
```

Comment:
This feels like it should be extracted from `Reconcile` in `main.go` so that the same loading code can be used here and in the `Reconcile` method. In fact, it seems like the `validate` command should be:

```golang
		RunE: func(cmd *cobra.Command, args []string) error {
			configFile, _ := cmd.Flags().GetString("config")
			config, restrictions, groups, err := ReadConfigFile(configFile)
			if err != nil {
				return err;
			}
			return Validate(config, restrictions, groups)
		},
```

Reply:
(and `plan` and `apply` would add a call to `Reconcile(print_only, config, groups)` to the end of the `RunE` method above.)